### PR TITLE
558 Added -pluginList and others to offcial doc

### PR DESCRIPTION
**Pull request by jpmartins**
Starting point was issue #558, used the -help and added the options that were missing.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [n/a] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: docs/running.rst [hunk 1]
```diff
@@ -135,6 +135,10 @@ These options may be used with both the GUI and command-line interfaces.
   Enable analyses which increase precision and find more bugs, but which may require more memory and take more time to complete.
   See :doc:`effort`.
 
+-effort[:less|default|more]:
```
Comment (KengoTODA) [inline docs/running.rst]: better to use `effort:[less|default|more]`?

File: docs/running.rst [hunk 2]
```diff
@@ -231,3 +235,110 @@ These options are only accepted by the Text User Interface.
...
+-release <release name>:
+  Set the release name of the analyzed application.
+
+-maxRank <rank>:
+  Only report issues with a bug rank at least as scary as that provided.
+
+-dontCombineWarnings:
+  Don't combine warnings that differ only in line number.
+
+-train[:outputDir]:
```
Comment (KengoTODA) [inline docs/running.rst]: maybe `train:[outputDir]` or `train <outputDir>`?

File: docs/running.rst [hunk 3]
```diff
@@ -231,3 +235,110 @@ These options are only accepted by the Text User Interface.
...
+  Save training data (experimental); output dir defaults to '.'.
+
+-useTraining[:inputDir]:
```
Comment (KengoTODA) [inline docs/running.rst]: `useTraining:[inputDir]` or `useTraining <inputDir>`?

**Comment by KengoTODA** [comment]
Really good, please change the target branch from `master` to `release-3.1`.

**Comment by jpmartins** [comment]
Hi, sorry for the hold up, not sure for all questions, i will look them up, as for changing target, as soon as i can i will search how to do that. Thanks

File: docs/running.rst (hunk 2 above)
Comment (jpmartins) [reply docs/running.rst]: It <is> optional, I found it [on the source].

```
class edu.umd.cs.findbugs.TextUICommandLine:
addSwitchWithOptionalExtraPart("-train", "outputDir", "Save training data (experimental); output dir defaults to '.'");
```

File: docs/running.rst (hunk 1 above)
Comment (jpmartins) [reply docs/running.rst]: Not sure. The less (-precision) and more (+precision) are intermediary levels between min (--precision) and max (++precision). Not sure about default...

**Comment by jpmartins** [comment]
Think i got it. Is it OK now?

File: docs/running.rst (hunk 1 above)
Comment (KengoTODA) [reply docs/running.rst]: I just meant that the position of `:` seems strange. 

BTW default value should be `more`, I investigated and noted at http://spotbugs.readthedocs.io/en/latest/effort.html

Comment (KengoTODA) [reply docs/running.rst]: In other words, we have five levels: min, less, default, more and max. And for now default and more has no difference. See `FindBugs` class's `XXX_EFFORT` fields for detail.

File: docs/running.rst (hunk 2 above)
Comment (KengoTODA) [reply docs/running.rst]: So `[:foo]` means optional parameter? It's first time for me to see this expression...

Comment (jpmartins) [reply docs/running.rst]: The The braces [ ] to indicate "zero or more", https://www.techrepublic.com/blog/software-engineer/what-is-bnf-and-why-should-developers-care 
Not sure about the colon : meaning yet, but guessing it is a variable indicator in some sort of notation.

File: docs/running.rst (hunk 1 above)
Comment (jpmartins) [reply docs/running.rst]: About the position of the : 
In CLI the context help ouputs:
-effort[:min|less|default|more|max]      set analysis effort level
Will re-write and propose to put together in on entry.

File: docs/running.rst (hunk 2 above)
Comment (KengoTODA) [reply docs/running.rst]: OK I see, you copied output from that command line help, then I can understand its meaning.

That `:` means necessary separator. For instance, for `-train[:outputDir]` means that we need to input like `-train:/tmp/foo`.

**Comment by KengoTODA** [comment]
This PR includes many unexpected commits, please check. You may recreate new PR with rebased branch.

**Comment by jpmartins** [comment]
Replaced by PR "Issue 558++". Not sure what i've done wrong to created unexpected commits, guessing it was because i started working on master.

**Pull request closed without merging**